packaging_automation/common_tool_methods.py

```diff
@@ -290,3 +321,114 @@ def remove_cloned_code(exec_path: str):
+    try:
+        cmd = f'gpg --batch --export-secret-keys -a "{fingerprint}" | base64'
+        ps = subprocess.run(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, timeout=2)
+        secret_key = ps.stdout.decode("ascii")
+        return secret_key
+    except subprocess.TimeoutExpired:
+        raise ValueError(
+            f"Error while getting key. Most probably packaging key is stored with password. "
+            f"Please remove the password when storing key with fingerprint {fingerprint}")
+
+
+def get_secret_key_by_fingerprint_with_password(fingerprint: str, passphrase: str) -> str:
+    try:
+        gpg = gnupg.GPG()
+        private_key = gpg.export_keys(fingerprint, True, passphrase=passphrase)
+
+        return base64.b64encode(private_key.encode("ascii")).decode("ascii")
+    except subprocess.TimeoutExpired:
+        raise ValueError(
+            f"Error while getting key. Most probably packaging key is stored with password. "
+            f"Please remove the password when storing key with fingerprint {fingerprint}")
+
+
+def get_public_gpg_key(fingerprint: str) -> str:
+    gpg = gnupg.GPG()
+    ascii_armored_public_keys = gpg.export_keys(fingerprint)
+    return base64.b64encode(ascii_armored_public_keys.encode("ascii")).decode("ascii")
+
+
+def define_rpm_public_key_to_machine(fingerprint: str):
+    run(f"gpg --export -a {fingerprint} >rpm_public.key")
+    run("rpm --import rpm_public.key")
+    os.remove("rpm_public.key")
+
+
+def delete_rpm_key_by_name(key_name: str):
+    result = subprocess.run(["rpm", "-q gpg-pubkey", "--qf %{NAME}-%{VERSION}-%{RELEASE}\t%{SUMMARY}\n"],
+                            stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    output = result.stdout.decode("ascii")
+    if output == "package gpg-pubkey is not installed":
+        print("There is not rpm public key to delete")
+    else:
+        key_lines = output.splitlines()
+        key_lines_filtered = filter(lambda line: key_name in line, key_lines)
+        for key_line in key_lines_filtered:
+            keys = key_line.split()
+            if len(keys) > 0:
+                run(f"rpm -e {keys[0]}")
+                print(f"{keys[0]} deleted")
+
+
+def is_rpm_file_signed(file_path: str) -> bool:
+    result = run_with_output(f"rpm -K {file_path}")
+    if result.returncode == 0:
+        return True
+    else:
+        return False
+
+
+def verify_rpm_signature_in_dir(rpm_dir_path: str):
+    files = list()
+    for (dirpath, dirnames, filenames) in os.walk(rpm_dir_path):
+        files += [os.path.join(dirpath, file) for file in filenames]
+    rpm_files = filter(lambda file_name: file_name.endswith("rpm"), files)
+    for file in rpm_files:
+        if not is_rpm_file_signed(f"{file}"):
+            raise ValueError(f"File {file} is not signed or there is a signature check problem")
+
+def remove_prefix(text, prefix):
+    return text[text.startswith(prefix) and len(prefix):]
```

Comment:
I think what is suggested is more understandable because it is simpler. For example what does it mean when `startswith` returns `False`? (I guess if we were to go with the suggestion, we wouldn't need to answer this)

Reply:
Fixed with a small change